server/session/event_rewrite_test.go

```
@@ -0,0 +1,354 @@
+		t.Fatalf("expected extra-event error before commit, got committed=%v err=%v", committed, err)
+	}
+	events, err := store.ReadEvents()
+	if err != nil {
+		t.Fatalf("read events: %v", err)
+	}
+	if len(events) != 1 || events[0].Kind != "message" {
+		t.Fatalf("expected no committed changes, got %+v", events)
+	}
+}
+
+type failingRewriteObserver struct {
```

golangci-lint (2.12.2)

[error] type failingRewriteObserver is unused

(unused)